### PR TITLE
Feat config

### PR DESCRIPTION
**Pull request by LaugeHermansen**
added simple hydra configs
added wandb logging
small fixes in docker container files

### CODE REVIEW

File: .devcontainer/gpu/devcontainer.json [hunk 1]
```diff
@@ -21,8 +21,9 @@
         "gpu": true
     },
     "runArgs": [
+        "--shm-size=4gb",
         "--gpus=all"
     ],
     "overrideCommand": true,
-    "postStartCommand": "pip install -e ."
+    "postStartCommand": "pip install -e .[dev]"
```
Comment (sarphiv) [inline .devcontainer/gpu/devcontainer.json:28]: Revert these changes. They're outside the scope of your issue and they are handled by us.

File: .devcontainer/cpu/devcontainer.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
             ]
         }
     },
-
+    "runArgs": ["--shm-size=4gb"],
     "overrideCommand": true,
-    "postStartCommand": "pip install -e ."
+    "postStartCommand": "pip install -e .[dev]"
```
Comment (sarphiv) [inline .devcontainer/cpu/devcontainer.json:22]: Revert these changes. They're outside the scope of your issue and they are handled by us.

File: src/ml_backend/train_model.py [hunk 3]
```diff
@@ -17,6 +18,10 @@
 import urllib.request
 from PIL import Image
```
Comment (sarphiv) [inline src/ml_backend/train_model.py:19]: This is built into Python and should be an import at the top. Sort your imports to be built-in, third-party, ours.

File: src/ml_backend/train_model.py [hunk 4]
```diff
@@ -27,7 +32,7 @@ def get_transform(model):
     return transform
 
 def get_dataloader(transform, split: Literal["train", "test"], batch_size: int, num_workers: int, **dataloader_kwargs) -> DataLoader:
-    dataset = torch.load(f"./data/processed/CIFAR10/{split}_dataset.pt")
+    dataset = torch.load(f"/workspace/data/processed/CIFAR10/{split}_dataset.pt")
```
Comment (sarphiv) [inline src/ml_backend/train_model.py:35]: Why absolute path?

File: src/ml_backend/train_model.py
Comment (sarphiv) [inline src/ml_backend/train_model.py:1]: Missing comments and documentation

**Review by sarphiv — CHANGES_REQUESTED** [review]

**Comment by sarphiv** [comment]
Also, in the future - add comment tying PR to issue #8

File: src/ml_backend/train_model.py [hunk 5]
```diff
@@ -1,68 +1,113 @@
...
-optimizer = model.configure_optimizers()
-
+@hydra.main(config_path="../../configs", config_name="config", version_base="1.3")
+def train(cfg):
+    """
+    train the model using the config file and hydra
+    """
+
+    # set seed
+    torch.manual_seed(cfg.seed)
```
Comment (sarphiv) [inline src/ml_backend/train_model.py:75]: Why not use pytorch lightning's seed everything?

File: src/ml_backend/train_model.py
Comment (davidAlvaBro) [inline src/ml_backend/train_model.py:1]: The number of workers in the dataloader is hard coded. Can we use the one in the config file?

File: src/ml_backend/train_model.py [hunk 6]
```diff
@@ -1,68 +1,113 @@
...
+
+    # instantiate the pl model
+    model = BaseModel(
+        timm_model,
+        learning_rate=cfg.models.learning_rate,
+        weight_decay=cfg.models.weight_decay,)
+
+    # instantiate the logger
+    logger = WandbLogger(
+        project="dtu_mlops_02476",
```
Comment (sarphiv) [inline src/ml_backend/train_model.py:96]: Should be a configuration setting

**Review by sarphiv — APPROVED** [review]
